client/me/account/main.jsx

```diff
@@ -84,25 +82,58 @@ const Account = React.createClass( {
 		debug( this.constructor.displayName + ' component is unmounting.' );
 	},
 
-	updateLanguage() {
-		let valueLink = this.valueLink( 'language' );
+	getUserSetting( settingName ) {
+		return this.props.userSettings.getSetting( settingName );
```

Comment:
Hmm, is there any rule-of-thumb to determine when the code should be defensive like this, and when to just trust the inputs?

After all, in a dynamic language like JS, one can never be sure about the type and shape of anything.

There is a lot of method calls on the `this.props.userSettings` and `this.props.username` in this module, so it would make sense to wrap them all in `invoke`.

The only place where this component is instantiated is the `controller.js` in the same directory. These two components are tightly-coupled "friends", so I think that the `client/me/account/main.jsx` component shouldn't need to worry if its props have the right type. What do you think?